src/ol/source/DataTile.js

```diff
@@ -228,6 +309,115 @@ class DataTileSource extends TileSource {
+   *
+   * @param {import("../proj.js").ProjectionLike} projection Projection.
+   * @param {import("../tilegrid/TileGrid.js").default} tilegrid Tile grid to use for the projection.
+   * @api
+   */
+  setTileGridForProjection(projection, tilegrid) {
+    const proj = getProjection(projection);
+    if (proj) {
+      const projKey = getUid(proj);
+      if (!(projKey in this.tileGridForProjection)) {
+        this.tileGridForProjection[projKey] = tilegrid;
+      }
+    }
+  }
+
+  /**
+   * @param {import("../proj/Projection.js").default} projection Projection.
+   * @return {import("../TileCache.js").default} Tile cache.
+   */
+  getTileCacheForProjection(projection) {
+    const thisProj = this.getProjection();
+    if (!thisProj || equivalent(thisProj, projection)) {
+      return this.tileCache;
+    } else {
+      const projKey = getUid(projection);
+      if (!(projKey in this.tileCacheForProjection)) {
+        this.tileCacheForProjection[projKey] = new TileCache(0.1); // don't cache
+      }
+      return this.tileCacheForProjection[projKey];
+    }
+  }
+
+  /**
+   * @param {import("../proj/Projection.js").default} projection Projection.
+   * @param {!Object<string, boolean>} usedTiles Used tiles.
+   */
+  expireCache(projection, usedTiles) {
+    const usedTileCache = this.getTileCacheForProjection(projection);
+
+    this.tileCache.expireCache(
+      this.tileCache == usedTileCache ? usedTiles : {}
+    );
+    for (const id in this.tileCacheForProjection) {
+      const tileCache = this.tileCacheForProjection[id];
+      tileCache.expireCache(tileCache == usedTileCache ? usedTiles : {});
+    }
+  }
+
+  clear() {
+    super.clear();
+    for (const id in this.tileCacheForProjection) {
+      this.tileCacheForProjection[id].clear();
+    }
+  }
 }
 
 export default DataTileSource;
+
+/**
+ * Determines the browser's ability to accurately reproject data. Some browsers
+ * (mostly Gecko-based) may interpolate byte values and corrupt Float32 values,
+ * especially when reprojecting between non-parallel projections, in which case
+ * an alternative output may be preferable.
+ * @return {boolean} True if the browser can accurately reproject data.
+ * @api
+ */
+export function getReprojectCapability() {
+  const ctx1 = createCanvasContext2D(2, 2, canvasPool);
+  const imageData = ctx1.getImageData(0, 0, 2, 2);
+  imageData.data[0] = 255;
+  imageData.data[3] = 255;
+  imageData.data[7] = 255;
+  imageData.data[11] = 255;
+  imageData.data[12] = 255;
+  imageData.data[15] = 255;
+  ctx1.putImageData(imageData, 0, 0);
+  const ctx2 = createCanvasContext2D(1, 1, canvasPool);
+  ctx2.imageSmoothingEnabled = false;
+  ctx2.drawImage(ctx1.canvas, 0, 0, 2, 2, 0, 0, 1, 1);
+  const data = ctx2.getImageData(0, 0, 1, 1).data;
+  releaseCanvas(ctx1);
+  canvasPool.push(ctx1.canvas);
+  releaseCanvas(ctx2);
+  canvasPool.push(ctx2.canvas);
+  return data[0] === 0 || data[0] === 255;
+}
```

Comment:
I don't think this is a function that we necessarily need or want to maintain in the API (better at the application level).  So I've pushed a commit that removes it.

Reply:
No problem with that.  I now have a workaround which reduces an image (by removing rows or columns) so it does not need to be downscaled but that is perhaps best left as a follow-up PR?